### PR TITLE
Added in supportability for network watcher and NSG flow logs

### LINKED ISSUE #410: Enabling NSG Flow logs will cause a dependency failure
**Describe the bug**
Enabling NSG Flow logs will cause a dependency failure

**To Reproduce**
1.In the Networking Details tab 
2. Enable "Limit Ingress/Egress subnets with Network Security Groups (NGS's)"
3. Enable "Capture NSG Flow Logs with Network Watcher"
4. Causes the dependency error below 
```
{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "ResourceGroupNotFound",
      "message": "Resource group 'NetworkWatcherRG' could not be found."
    }
  ]
}
```

**Expected behavior**
Enabling both NSG option would not cause a dependency issue.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by MattLeach25**
## PR Summary

When enabling NSG flow logs, the pipeline would fail stating that NetworkWatcherRG was not present. This resource group wasn't created at the time the NSG flow logs were getting enabled. 

Added in compatibility to support the user to create the resource group first and added in network watcher resource before the flow logs get enabled. This has fixed the issue.

Closes #410

image

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] This PR is ready to merge and is not **Work in Progress**
- [x] Link to a filed issue
- [x] Screenshot of UI changes (if PR includes UI changes)

### CODE REVIEW

**Comment by Gordonby** [comment]
Do you want the baseline changes in here, or are they for another pr?

**Comment by Gordonby** [comment]
> > Do you want the baseline changes in here, or are they for another pr?
> 
> @Gordonby I don't mind the baseline changes being in there because they're just key value changes. I have however changed the disabled value back to true, so users will not see this card until we've finished it.
> 
> There are a few pipeline failures though so need to look into them.

Yep, moving to disabled is fine.

GitHub says that you'll need to update your branch from main, so do that before worrying too much about Ci failures.

**Comment by Gordonby** [comment]
Broken link detector isn't letting this through 😄😄😄

Looks like the coreos domain is gone.

> ## Errors per input
> ### Errors in helper/localsite.html
>* [https://coreos.com/operators/prometheus/docs/latest/user-guides/getting-started.html]>(https://coreos.com/operators/prometheus/docs/latest/user-guides/getting-started.html): Failed: Network error (status code: ERR)

Can you swap it out with this one please?
https://prometheus-operator.dev/docs/user-guides/getting-started/

File: bicep/networkwatcherflowlog.bicep [hunk 1]
```diff
@@ -12,6 +12,11 @@ param workspaceId string = ''
 param workspaceResourceId string = ''
 param workspaceRegion string = resourceGroup().location
 
+resource networkWatcher 'Microsoft.Network/networkWatchers@2022-01-01' = {
```
Comment (Gordonby) [inline bicep/networkwatcherflowlog.bicep:15]: Do we need to worry about the scope that this is run at?
It definitely creates in the right resource group?

File: helper/src/components/deployTab.js [hunk 2]
```diff
@@ -220,9 +220,13 @@ export default function DeployTab({ defaults, updateFn, tabValues, invalidArray,
     post_deploycmd.replaceAll('"', '\\"') +
     `\n"${deploy.selectedTemplate === "local" ? ' --file ./postdeploy/scripts/postdeploy.sh --file ./postdeploy/helm/Az-CertManagerIssuer-0.3.0.tgz --file ./postdeploy/k8smanifests/networkpolicy-deny-all.yml --file ./helper/src/dependencies.json' : ''}`
 
+  const networkWatcher = net.nsg && net.nsgFlowLogs !== defaults.net.nsgFlowLogs ?
+    `# Create Network Watcher Resource Group If It Doesn't Exist\n ` +
```
Comment (Gordonby) [inline helper/src/components/deployTab.js:224]: if you can get rid of this one character of whitespace - then we're good to approve/merge @MattLeach25

Comment (MattLeach25) [reply helper/src/components/deployTab.js]: Resolved.

**Review by Gordonby — APPROVED** [review]
lgtm

**Pull request merged**